apps/rule-manager/client/src/ts/components/hooks/useBatchRules.ts

```diff
@@ -0,0 +1,93 @@
+import {useEffect, useState} from "react";
+import { errorToString } from "../../utils/error";
+import {DraftRule, RuleData} from "./useRule";
+import {ErrorIResponse, responseHandler} from "../../utils/api";
+
+export function useBatchRules(ruleIds: number[] | undefined) {
+  const [isLoading, setIsLoading] = useState(false);
+  const [errors, setErrors] = useState<string | undefined>(undefined);
+  const [rules, setRules] = useState<RuleData[] | undefined>(undefined);
+
+  const fetchRules = async (ruleIds: number[]) => {
+    setIsLoading(true);
+
+    try {
+      const response = await fetch(`${location}rules/batch/${ruleIds.join(',')}`);
+      if (!response.ok) {
+        throw new Error(`Failed to fetch rules: ${response.status} ${response.statusText}`);
+      }
+      const rulesData: RuleData[] = await response.json();
+      setRules(rulesData)
+
+    } catch (error) {
+      setErrors(errorToString(error));
+    }
+
+    setIsLoading(false);
+  }
+
+  const updateRules = async (ruleForm: DraftRule[]) => {
+    setIsLoading(true);
+
+    const hasCategoryChanged = rules && rules[0].draft.category !== ruleForm[0].category;
+
+    const haveTagsChanged = () => {
+        if (!rules) return false;
+        const oldTags = rules.map(rule => rule.draft.tags).flat()
```

Comment:
Not blocking, but this (and other instances in this file) is equivalent to `flatMap` –
```suggestion
        const oldTags = rules.flatMap(rule => rule.draft.tags)
```

Reply:
Good spot, thanks for that!